### PR TITLE
Accept a higher native timeout in install os for nexus.

### LINKED ISSUE #232: N5Ks can be slow the native timeout should be configurable.
https://github.com/networktocode/pyntc/blob/b46b94a202c1dc5bf30c97dcaa69af8fe77bde60/pyntc/devices/nxos_device.py#L365

Constantly hitting the `requests` module `read timeout` which is 300 on N5K installs.  These installs go through the validation of both the system and kickstart image.  Regularity takes longer than 10 minutes to just finish the "verification" steps.  This should be configurable to avoid timeout issues.

### PR DESCRIPTION
**Pull request by jeffkala**
fixes #232

### CODE REVIEW

File: pyntc/devices/nxos_device.py [hunk 1]
```diff
@@ -362,7 +362,8 @@ def set_boot_options(self, image_name, kickstart=None, **vendor_specifics):
             kickstart = file_system + kickstart
 
         image_name = file_system + image_name
-        self.native.timeout = 300
+        if self.native.timeout < 300:
```
[the remark this answers is not in the document]
Comment (jeffkala) [reply pyntc/devices/nxos_device.py:366]: Added a comment.  Essentially just to support a user defined timeout if its higher than 300.  I'd say if a user is defining a higher timeout it is for a reason.  Slow switch, or in my case longer upgrade cycle for n5k hardware.

**Review by mundruid — APPROVED** [review]
Thank you!

**Comment by jeffkala** [comment]
Let me know if anything else is needed.

File: pyntc/devices/nxos_device.py [hunk 2]
```diff
@@ -362,7 +362,9 @@ def set_boot_options(self, image_name, kickstart=None, **vendor_specifics):
             kickstart = file_system + kickstart
 
         image_name = file_system + image_name
-        self.native.timeout = 300
+        # Allow for user defined timeout to take precedence if its over 300 seconds, otherwise change to 300.
+        if self.native.timeout < 300:
```
Comment (jmcgill298) [inline pyntc/devices/nxos_device.py:366]: what if self.native.timeout isn't an int?

**Review by jmcgill298 — CHANGES_REQUESTED** [review]

File: pyntc/devices/nxos_device.py (hunk 2 above)
Comment (jeffkala) [reply pyntc/devices/nxos_device.py]: This has been updated @jmcgill298

**Review by mundruid — APPROVED** [review]

**Pull request merged**